### PR TITLE
chore(pkglint): don't allow adding peer dependencies to monocdk

### PR DESCRIPTION
**Pull request by NetaNir**
Adding a peer dependency to `monocdk` is a breaking change. Added a lint rule that only allows `constructs` as a peer dependency of monocdk.

This PR build is currently failing since `cdk8s` was added as a peer dependency of  `monocdk`, thus breaking the rule. Once it is removed this PR will build successfully. See related [discussion](https://github.com/aws/aws-cdk/pull/10562/files#r505074526).

From the build logs:
```

monocdk: In package package.json
--
963 | monocdk: - [monocdk/no-peer] Adding a peer dependency to monocdk is a breaking change, and thus not allowed.
964 | monocdk:          Added cdk8s
965 | monocdk: Error: Some package.json files had errors
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/pkglint/lib/rules.ts [hunk 1]
```diff
@@ -487,6 +487,26 @@ export class JSIIProjectReferences extends ValidationRule {
   }
 }
 
+export class NoPeerDependenciesMonocdk extends ValidationRule {
+  public readonly name = 'monocdk/no-peer';
+  public allowedPeer = ['constructs'];
+
+  public validate(pkg: PackageJson): void {
+    if (pkg.packageName !== 'monocdk') {
```
Comment (eladb) [inline tools/pkglint/lib/rules.ts:495]: what about aws-cdk-lib?

Comment (NetaNir) [reply tools/pkglint/lib/rules.ts:495]: aws-cdk-lib is a private package which is under active development.

Comment (nija-at) [reply tools/pkglint/lib/rules.ts:495]: We should include aws-cdk-lib to this and update the error message.

File: tools/pkglint/lib/rules.ts [hunk 2]
```diff
@@ -487,6 +487,26 @@ export class JSIIProjectReferences extends ValidationRule {
...
 }
 
+export class NoPeerDependenciesMonocdk extends ValidationRule {
+  public readonly name = 'monocdk/no-peer';
+  public allowedPeer = ['constructs'];
+
+  public validate(pkg: PackageJson): void {
+    if (pkg.packageName !== 'monocdk' && pkg.packageName !== 'aws-cdk-lib') {
+      return;
+    }
```
Comment (eladb) [inline tools/pkglint/lib/rules.ts:497]:
```suggestion
  private readonly allowedPeer = ['constructs'];
  private readonly modules = ['monocdk', 'aws-cdk-lib'];

  public validate(pkg: PackageJson): void {
    if (!modules.includes(pkg.packageName)) {
      return;
    }
```

**Review by eladb — APPROVED** [review]

**Comment by eladb** [comment]
added `do-not-merge`

**Pull request merged**